Source/WebCore/rendering/RenderTreeAsText.cpp

```diff
@@ -363,59 +363,35 @@ void RenderTreeAsText::writeRenderObject(TextStream& ts, const RenderObject& o,
         }
         if (borderTop || borderRight || borderBottom || borderLeft) {
             ts << " [border:";
-
-            BorderValue prevBorder = o.style().borderTop();
-            if (!borderTop)
-                ts << " none";
-            else {
-                ts << " (" << borderTop << "px ";
-                printBorderStyle(ts, o.style().borderTopStyle());
-                auto color = o.style().borderTopColor();
-                if (!color.isValid())
-                    color = o.style().color();
-                ts << serializationForRenderTreeAsText(color) << ")";
-            }
+            BorderValue prevBorder;
+
+            auto printBorderStyleAndColor = [&ts, &o, &prevBorder] 
+                (BorderValue const& currentBorder, LayoutUnit const& border, BorderStyle const& style, Color color)
+                {
```

Comment:
```suggestion
            auto printBorder = [&ts, &o] (LayoutUnit const& width, BorderStyle const& style, Color color) {
```

4 suggestions:
* Maybe rename `border` to `width` (since that's what it actually represents)
* This could be named `printBorder` (since it prints all width as well)
* I would move out the `prevBorder = currentBorder;` logic (even if it's a bit more duplication), since it's not directly related to printing.
* I think it's more readable on a single line